src/System.Collections.NonGeneric/tests/Performance/Perf.ArrayList.cs

```diff
@@ -0,0 +1,476 @@
+            {
+                arrayListOfInts.Add(element);
+            }
+
+            return arrayListOfInts;
+        }
+
+        [Benchmark(InnerIterationCount = 2000000)]
+        public void Add()
+        {
+            int innerIterationCount = (int)Benchmark.InnerIterationCount;
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                var elements = new ArrayList();
+                using (iteration.StartMeasurement())
+                {
+                    for (int element = 0; element < innerIterationCount; element++)
+                    {
+                        elements.Add(element);
+                    }
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = 8000)]
+        public void AddRange()
+        {
+            int innerIterationCount = (int)Benchmark.InnerIterationCount;
+            ArrayList elements = CreateArrayListOfInts(10000);            
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int element = 0; element < innerIterationCount; element++)
+                    {
+                        var elementsCollection = new ArrayList();
+                        elementsCollection.AddRange(elements);
+                    }                    
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = 10)]
+        public void BinarySearch()
+        {
+            int innerIterationCount = (int)Benchmark.InnerIterationCount;
+            ArrayList elements = CreateArrayListOfInts(100000);
+            Comparer<int> comparer = Comparer<int>.Default;
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int index = 0; index < innerIterationCount; index++)
+                    {
+                        for (int i = 0; i < elements.Count; i++)
+                        {
+                            elements.BinarySearch(i, comparer);
+                        }
+                    }                    
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = 5000)]
+        public void Clear()
+        {
+            int innerIterationCount = (int)Benchmark.InnerIterationCount;
+            ArrayList elements = CreateArrayListOfInts(10000);
+            int elementsCollectionSize = (int)Benchmark.InnerIterationCount;
+            foreach (var iteration in Benchmark.Iterations)
+            {              
+                ArrayList[] elementsCollection = new ArrayList[elementsCollectionSize];
+                for (int index = 0; index < elementsCollectionSize; index++)
+                {
+                    elementsCollection[index] = new ArrayList(elements);
+                }                    
```

Comment:
I wonder if it's worthwhile doing similar prep for the AddRange case earlier.  Might not be worthwhile given that in that case the prep is just the allocation/ctor for the temporary ArrayList.

Reply:
That's not bad idea. I am fixing the AddRange test.